code/_core/obj/item/magazine/_magazine.dm

```diff
@@ -21,10 +21,32 @@
 	var/bullet_diameter_max = -1
 
 	var/icon_states = 1
+	var/bluespaced = FALSE
+	var/regenerate = FALSE
+	var/regen_counter = 0
 
 	weight = 0.25
 
 
+/obj/item/magazine/PostInitialize()
+	start_thinking(src)
+	return ..()
+
+/obj/item/magazine/think()
+	if (regenerate)
+		regen_counter += 1
+		if (regen_counter > 30)
+			regen_counter = 0
+			if (length(stored_bullets) < bullet_count_max)
+				var/obj/item/bullet_cartridge/B = new ammo(src)
+				INITIALIZE(B)
+				FINALIZE(B)
```

Comment:
GENERATE(B) should be placed between INITIALIZE(B) and FINALIZE(B) as an object is being generated for the first time.